docs/developing-contracts/examples/DAO.md

```diff
@@ -0,0 +1,196 @@
+---
+sidebar_label: 'DAO'
+sidebar_position: 3
+---
+
+# Decentralized autonomous organization
+
+The smart-contract is avaialbe on GitHub: https://github.com/gear-tech/apps/tree/master/dao
+
+### Structs
+
+```rust
+/// Initial parameters:
+struct InitConfig {
+    admin: H256,
+    approved_token_program_id: H256,
+    guild_bank_id: H256,
+    period_duration: u64,
+    voting_period_length: u64,
+    grace_period_length: u64,
+    dilution_bound: u128,
+    abort_window: u64,
+}
+```
+
+`admin` - the only member that is initially in DAO. He has 1 share that allows him to submit new proposal and therefore add new members.
+
+`approved_token_program_id` - the reference to the token contract (ERC20) that users use as pledge to get the share in the DAO.
+
+`period_duration` - the smallest unit time interval for the DAO.
+
+`voting_period_length` - voting time interval.
+
+`grace_period_length` - after the voting period the DAO members are given a period of time in which they can leave the DAO(ragequit) without being diluted and ultimately affected by the proposal’s acceptance into the DAO.
+
+`dilution_bound` - The dilution bound protects members from excessive dilution in case of a mass ragequit. It is designed to mitigate an issue where a proposal is passed, then many users rage quit from the DAO. The proposal will be automatically rejected if the total amount of shares becomes dilutionBound times less than it was before.
+
+`abort_window` - the time interval during which the applicant can cancel the proposal with funds refunded (that starts immediately when the proposal is submitted).
+
+```rust
+struct Dao {
+    admin: ActorId,
+    approved_token_program_id: ActorId,
+    period_duration: u64,
+    voting_period_length: u64,
+    grace_period_length: u64,
+    dilution_bound: u128,
+    abort_window: u64, 
+    total_shares: u128,
+    members: BTreeMap<ActorId, Member>,
+    member_by_delegate_key: BTreeMap<ActorId, ActorId>,
+    proposal_id: u128,
+    proposals: BTreeMap<u128, Proposal>,
+ 	approved_list: Vec<ActorId>,
```

Comment:
Seem there are some extra spaces in this line.

Reply:
fixed